llvm/include/llvm/ObjectYAML/DXContainerYAML.h

```diff
@@ -264,6 +267,14 @@ template <> struct MappingTraits<DXContainerYAML::RootSignatureYamlDesc> {
                       DXContainerYAML::RootSignatureYamlDesc &RootSignature);
 };
 
+template <> struct MappingTraits<dxbc::RootParameter> {
```

Comment:
Once you rework the RootParameters in the BinaryFormat headers this is going to need to be fixed too.

For all the other data structures we've encoded to and from YAML we have not used the definitions from the BinaryFormat headers to allow these to be simpler.

Declaring our own structures for these allows us to do things like using `std::variant`, and to generally not be bound tightly to the object's binary encoding scheme.